kobo/apps/subsequences/models.py

```diff
@@ -194,6 +194,12 @@ def retrieve_data(
                     # where ('translation','lang1') would be one key and
                     # ('translation', 'lang2') would be the other
                     transformed_data = action.transform_data_for_output(retrieved_data)
+                    if action.returns_structured_output():
```

Comment:
Let's keep this generic

```suggestion
                    if not action.overlaps_other_actions():
                        for field_key, field_data in transform_data.items():
                             if field_data:
                                  output_data_for_question[field_key] = field_data
                        continue
```